traits/src/router.rs

```diff
@@ -3,9 +3,14 @@ use frame_support::dispatch::DispatchResultWithPostInfo;
 use frame_support::sp_runtime::{DispatchError, DispatchResult};
 use frame_support::weights::Weight;
 use scale_info::TypeInfo;
+use sp_arithmetic::FixedU128;
 use sp_std::vec;
 use sp_std::vec::Vec;
 
+pub trait RouteSpotPriceProvider<AssetId> {
+	fn spot_price(route: &[Trade<AssetId>]) -> Option<FixedU128>;
+}
+
```

Comment:
not a fan of new price provider trait. 

not sure what is the requirement here - but i believe you can use PriceProvider trait. 

i feel we have some underlying issue due to the fact that we always need to create new price provider traits.